controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -305,6 +318,203 @@ func (r *AutoscalingRunnerSetReconciler) deleteEphemeralRunnerSets(ctx context.C
 	return nil
 }
 
+func (r *AutoscalingRunnerSetReconciler) removeFinalizersFromPermissionResources(ctx context.Context, autoscalingRunnerSet *v1alpha1.AutoscalingRunnerSet, logger logr.Logger) (requeue bool, err error) {
+	requeue, err = r.removeFinalizersContainerModeKubernetesPermissions(ctx, autoscalingRunnerSet, logger)
+	if err != nil {
+		return true, err
```

Comment:
let's add some trace before each return, so we know which code path it runs during debug.

Reply:
With this new cleanup, I added error message that is descriptive so we can know where the failure is happening. Good call